src/output.rs

```diff
@@ -55,12 +56,23 @@ impl IsPredicate {
 
         if result != self.expected_result {
             if self.expected_result {
-                bail!(ErrorKind::BytesDoesntMatch(
-                    expect.to_owned(),
-                    got.to_owned(),
-                ));
+                Err(AssertionError::new(AssertionKind::OutputMismatch))
+                    .context("expected to match")
+                    .context(KeyValueDisplay::new(
+                        "expect",
+                        DebugDisplay::new(expect.to_owned()),
+                    ))
+                    .context(KeyValueDisplay::new(
+                        "got",
+                        DebugDisplay::new(got.to_owned()),
+                    ))?;
```

Comment:
These 10 lines here (and below a few times) should be rewritten before landing this. I don't see why this should not be a constructor on a custom error – we are dealing with `expected` and `got` all over the place, and you are partially writing the display output here every time.